src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+
+/**
+ * Handle database data and operations for Duke.
+ * Source of truth for the task list as well as able to write to and read from
+ * the database as necessary.
+ */
+public class Storage {
+
+    /**
+     * Checks for the save file, and if it exists, read from it to restore tasks.
+     * Relies on the Task class to handle conversions to and from a task object to
+     * a string stored in the database.
+     */
+    public TaskList readFromDatabase() {
+        String dukeLocation = System.getProperty("user.dir");
+        Path filePath = Paths.get(dukeLocation, "data", "tasks");
+
+        if (!Files.isRegularFile(filePath)) {
+            return new TaskList(new ArrayList<>());
+        }
+
+        System.out.println("Reading last save  - " + filePath);
+        ArrayList<String> failedParses = new ArrayList<>();
+
+        ArrayList<Task> taskArr = new ArrayList<>();
+        try {
+            Stream<String> databaseToString = Files.lines(filePath);
+            databaseToString.forEach(
+                taskString -> {
+                    try {
+                        Task task = Task.stringToTask(taskString);
+                        taskArr.add(task);
+                    } catch (IllegalArgumentException e) {
+                        failedParses.add(taskString);
+                    }
+                }
+            );
+            databaseToString.close();
```

Comment:
Do you think this part can extract out as a method?

Reply:
Yep, makes sense. Thanks!